### ISSUE TITLE
No such file or directory returned when running for wildcard path

### ISSUE THREAD

**Issue by minherz** [adrienverge/yamllint#577]
When running `yamllint` with a command:

```bash
yamllint data/**/*.yaml
```
it returns an error:

```terminal
No such file or directory: 'data/**/*.yaml'
```

I would prefer to opt-in for this kind of error, at least when I run it for a wildcarded path. Absent file is not the same as invalid YAML. To use the tool I need a way to fail on linting errors and ignore everything else.

**Comment by andrewimeson** [comment]
This is expected. When that glob works, it's because the shell is expanding it. When the shell can't expand it because it doesn't match it is passing it to yamllint literally. Yamllint **shouldn't** treat it as a glob because it's a valid filename, and we aren't a shell.

```bash
$ mkdir -p data/aa/bb/cc
$ shopt -s globstar
$ echo 'glob_will_match: yes' >> data/aa/bb/cc/test.yaml
$ yamllint data/**/*.yaml
data/aa/bb/cc/test.yaml
  1:1       warning  missing document start "---"  (document-start)
  1:18      warning  truthy value should be one of [false, true]  (truthy)
$ rm -rf data/aa/
$ yamllint data/**/*.yaml
[Errno 2] No such file or directory: 'data/**/*.yaml'

$ mkdir data/\*\*
$ echo 'same_thing: yep ' >> data/\*\*/\*.yaml
$ tree
.
└── data
    └── **
        └── *.yaml

$ yamllint data/**/*.yaml
data/**/*.yaml
  1:1       warning  missing document start "---"  (document-start)
  1:16      error    trailing spaces  (trailing-spaces)
```

`*` is a legitimate character for a filename. If you're using bash and you don't want this behavior, you can set the `nullglob` option, but that will just result in running yamllint with no arguments.

```bash
$ rm -rf data/*
$ shopt -s nullglob
$ yamllint data/**/*.yaml
usage: yamllint [-h] [-] [-c CONFIG_FILE | -d CONFIG_DATA] [--list-files] [-f {parsable,standard,colored,github,auto}] [-s] [--no-warnings] [-v] [FILE_OR_DIR ...]
yamllint: error: one of the arguments FILE_OR_DIR - is required
```

**Comment by andrewimeson** [comment]
Some alternatives

```sh
find data/ -name '*.yaml' | xargs yamllint
```
or 
```sh
find data/ -name '*.yaml' -exec yamllint {} +
```

**Comment by minherz** [comment]
If I am not mistaken the first alternative will be the same as calling `yamllint` without arguments. But the second one should work, thank you.
